Sources/AsyncQueue/ActorQueue.swift

```diff
@@ -153,9 +161,13 @@ public final class ActorQueue<ActorType: Actor>: @unchecked Sendable {
     }
 }
 
-/// A global actor used for synchronizing task execution.
-@globalActor
-private struct ActorQueueSynchronization {
-    fileprivate actor Synchronization {}
-    fileprivate static let shared = Synchronization()
+extension Actor {
+    nonisolated
+    func execute(
+        _ task: @escaping @Sendable (isolated Self?) async -> Void
```

Comment:
a bit tacky but enables us to isolate the closure to the actor's context without retaining the actor.